controllers/kustomization_controller.go

```diff
@@ -688,10 +703,10 @@ func (r *KustomizationReconciler) getServiceAccountToken(kustomization kustomize
 	return token, nil
 }
 
-func (r *KustomizationReconciler) apply(kustomization kustomizev1.Kustomization, dirPath string) (string, error) {
+func (r *KustomizationReconciler) apply(ctx context.Context, kustomization kustomizev1.Kustomization, dirPath string) (string, error) {
 	start := time.Now()
 	timeout := kustomization.GetTimeout() + (time.Second * 1)
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	applyCtx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Why not use the context passed to the function to create the timeout context?

Reply:
Because we want to respect the timeout set in spec.timeout for apply, otherwise it could hand forever

Reply:
I think i did a bad job explaining my thoughts.

Instead of doing the first why not do the seconds?
```
applyCtx, cancel := context.WithTimeout(context.Background(), timeout)
```
vs.
```
applyCtx, cancel := context.WithTimeout(ctx, timeout)
```

Reply:
Ahh yes that makes sense, thanks 👍